hydromt/models/components/grid.py

```diff
@@ -13,461 +10,194 @@
-        -------
-        list
-            Names of added model map layers
+        data: xarray.DataArray or xarray.Dataset
+            new map layer to add to grid
+        name: str, optional
+            Name of new map layer, this is used to overwrite the name of a DataArray
+            and ignored if data is a Dataset
         """
-        rename = rename or {}
-        self.logger.info(f"Preparing grid data from raster source {raster_fn}")
-        # Read raster data and select variables
-        ds = self.data_catalog.get_rasterdataset(
-            raster_fn,
-            geom=self.region,
-            buffer=2,
-            variables=variables,
-            single_var_as_array=False,
-        )
-        # Data resampling
-        ds_out = workflows.grid.grid_from_rasterdataset(
-            grid_like=self.grid,
-            ds=ds,
-            variables=variables,
-            fill_method=fill_method,
-            reproject_method=reproject_method,
-            mask_name=mask_name,
-            rename=rename,
+        self._initialize_grid()
+        # NOTE: variables in a dataset are not longer renamed as used to be the case in
+        # set_staticmaps
+        name_required = isinstance(data, np.ndarray) or (
+            isinstance(data, xr.DataArray) and data.name is None
         )
-        # Add to grid
-        self.set_grid(ds_out)
+        if name is None and name_required:
+            raise ValueError(f"Unable to set {type(data).__name__} data without a name")
+        if isinstance(data, np.ndarray):
+            if data.shape != self._data.raster.shape:
+                raise ValueError("Shape of data and grid maps do not match")
+            data = xr.DataArray(dims=self._data.raster.dims, data=data, name=name)
+        elif isinstance(data, xr.DataArray):
+            if name is not None:
+                data.name = name
+            data = data.to_dataset()
+        elif not isinstance(data, xr.Dataset):
+            raise ValueError(f"cannot set data of type {type(data).__name__}")
 
-        return list(ds_out.data_vars.keys())
+        if len(self._data) == 0:  # empty grid
+            self._data = data
+        else:
+            for dvar in data.data_vars:
+                if dvar in self._data:
+                    if self.model_root.is_reading_mode():
+                        self.logger.warning(f"Replacing grid map: {dvar}")
+                self._data[dvar] = data[dvar]
 
-    def setup_grid_from_raster_reclass(
+    def write(
         self,
-        raster_fn: Union[str, Path, xr.DataArray],
-        reclass_table_fn: Union[str, Path, pd.DataFrame],
-        reclass_variables: List,
-        variable: Optional[str] = None,
-        fill_method: Optional[str] = None,
-        reproject_method: Optional[Union[List, str]] = "nearest",
-        mask_name: Optional[str] = "mask",
-        rename: Optional[Dict] = None,
+        fn: str = "grid/grid.nc",
+        temp_data_dir: Optional[StrPath] = None,
```

Comment:
Could this still be a ``Model`` property like before instead of a new argument? 
It's really to avoid PermissionError when writing the model. The argument should be part of ``write_nc`` as it is not linked to ``Model`` anymore but ``GridComponent`` still is.

```suggestion
```